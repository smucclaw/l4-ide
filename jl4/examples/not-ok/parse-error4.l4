ยง `The factorial function`

-- Report single error, even though we could see there are two issues
-- Namely MEANS and a missing 'ELSE' branch.
-- Error recovery only happens over 'topdecl'
GIVEN n IS A NUMBER
GIVETH A NUMBER
DECIDE factorial MEANS n
  IS
    IF n EQUALS 0
    THEN 1
