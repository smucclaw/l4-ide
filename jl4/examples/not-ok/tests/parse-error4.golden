  |
8 | DECIDE factorial MEANS n
  |                  ^^^^^
unexpected MEANS
expecting AKA, IF, IS, OF, identifier, or space token

